tests/post/steps/test_liveness.py

```diff
@@ -34,3 +45,43 @@ def check_exec(host, command, pod, namespace):
 
     with host.sudo():
         host.check_output(cmd)
+
+
+@then(
+    parsers.parse(
+        "we have {expected_nb_pods:d} running pod labeled '{label}'"
+    )
+)
+def count_running_pods(host, expected_nb_pods, label):
+    cmd = (
+        'kubectl --kubeconfig=/etc/kubernetes/admin.conf '
+        'get pod -l "{label}" --field-selector=status.phase=Running '
+        '--namespace kube-system -o json'
+    ).format(label=label)
+
+    # Just after the deployment, the pods may not in running state yet
+    # so implement a timeout
+    interval = 3
+    attempts = 10
+
+    with host.sudo():
+        for _ in range(attempts):
+
+            ret = host.run(cmd)
+            assert ret.rc == 0, ret.stdout
+
+            pods = json.loads(ret.stdout)
+
+            if len(pods['items']) < expected_nb_pods:
+                time.sleep(interval)
+            else:
+                break
+
+        else:
```

Comment:
Why is this an `else` block?

Reply:
This is a `else` of the `for` block. Without the `else` we don't know if the loop ends because of the break (pods are up) or because we reach the max attempts (pods still down)